plugin/storage/grpc/factory_test.go

```diff
@@ -143,6 +147,34 @@ func TestGRPCStorageFactory(t *testing.T) {
 	assert.Equal(t, f.store.DependencyReader(), depReader)
 }
 
+func TestGRPCStorageFactoryWithConfig(t *testing.T) {
+	cfg := grpcConfig.Configuration{}
+	_, err := NewFactoryWithConfig(cfg, metrics.NullFactory, zap.NewNop())
+	require.ErrorContains(t, err, "grpc-plugin builder failed to create a store: error connecting to remote storage")
+
+	lis, err := net.Listen("tcp", ":0")
+	require.NoError(t, err, "failed to listen")
+
+	s := grpc.NewServer()
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			log.Fatalf("Server exited with error: %v", err)
+		}
+	}()
+	defer s.Stop()
+
+	cfg.RemoteServerAddr = lis.Addr().String()
+	cfg.RemoteConnectTimeout = 1 * time.Second
+	f, err := NewFactoryWithConfig(cfg, metrics.NullFactory, zap.NewNop())
+	defer func() {
+		err := f.Close()
+		if err != nil {
+			log.Fatalf("Client exited with error: %v", err)
+		}
+	}()
+	require.NoError(t, err)
```

Comment:
incorrect order - when creating a closable object, you first need to check that no error was returned from constructor, and only the do defer/close (in this case you don't need defer since it's the last line of the function).

Reply:
> in this case you don't need defer since it's the last line of the function

Ah yes, that is a good point.